output/types.go

```diff
@@ -95,3 +96,9 @@ type WithRunStatusUpdates interface {
 	Output
 	SetRunStatus(latestStatus lib.RunStatus)
 }
+
+// WithBuiltinMetrics means the output can receive the builtin metrics.
+type WithBuiltinMetrics interface {
+	Output
+	SetBuiltinMetrics(builtinMetrics *metrics.BuiltinMetrics)
+}
```

Comment:
I'm not sure this is a pattern we want to use for many things, just like we want to avoid embedding everything in `Context`. In this case since `InitEnvironment` is already embedded in the `Context`, and we could potentially access the built-in metrics from `Registry`, what do you think about changing the `Output` interface and simply passing the context from the `Engine` to each output in `Start(ctx)`?

Reply:
InitEnvironment is embeded in VUs not in the Engine, so this will be quite a big change and also we do not provide the Output with a context to begin with

Reply:
Hhmm right, it's not the same context, and the `InitEnvironment` is embedded in `Bundle.instantiate()`, so the `Engine` doesn't have access to it.

> also we do not provide the Output with a context to begin with

I know, I was suggesting we should change `Output.Start()` to take one. That seems cleaner than adding another `With*` method to outputs, but not if it requires a lot of changes. So it's fine as is.